odl/space/fspace.py

```diff
@@ -899,72 +750,67 @@ def ipow_posint(x, n):
                 x *= tmp
                 return x
 
-        def power_call_out_of_place(x):
+        def power_oop(x, **kwargs):
             """Power out-of-place evaluation function."""
             if p == 0:
-                return self.one()
-            elif p == int(p) and p >= 1:
-                return np.asarray(pow_posint(x_call_oop(x), int(p)),
-                                  dtype=self.out_dtype)
-            else:
-                return np.power(x_call_oop(x), p).astype(self.out_dtype)
-
-        def power_call_in_place(x, out):
-            """Power in-place evaluation function."""
-            if p == 0:
-                out.assign(self.one())
-
-            x_call_ip(x, out)
-            if p == int(p) and p >= 1:
-                return ipow_posint(out, int(p))
+                return self.one()
+            elif p == int(p) and p >= 1:
+                return np.asarray(pow_posint(f_copy(x, **kwargs), int(p)),
+                                  dtype=self.scalar_out_dtype)
             else:
-                out **= p
-                return out
+                result = np.power(f_copy(x, **kwargs), p)
+                return result.astype(self.scalar_out_dtype)
 
-        out._call_out_of_place = power_call_out_of_place
-        out._call_in_place = power_call_in_place
-        out._call_has_out = out._call_out_optional = True
+        out._call_out_of_place = power_oop
+        decorator = preload_first_arg(out, 'in-place')
+        out._call_in_place = decorator(_default_in_place)
+        out._call_has_out = out._call_out_optional = False
         return out
 
-    def _realpart(self, x):
-        """Function returning the real part of a result."""
-        x_call_oop = x._call_out_of_place
+    def _realpart(self, f):
+        """Function returning the real part of the result from ``f``."""
+        # Avoid infinite recursions by making a copy of the function
```

Comment:
Happened sometime along the way, but with this implementation the problem doesn't occur any more. I'll remove this extra stuff.